### ISSUE TITLE
trainNMF error 

### ISSUE THREAD

**Issue by Lihyunseo** [MarcElosua/SPOTlight#116]
Hello! I am a student currently studying single-cell genomics and have been deeply impressed by your excellent model.

I tried following the vignettes/SPOTlight_kidney.Rmd that you posted, but encountered an error.

Here is the code I used:

mod_ls <- trainNMF(
  x = sce,
  y = spe,
  groups = sce$type,
  mgs = mgs,
  weight_id = "weight",
  group_id = "type",
  gene_id = "gene"
)

And here is the error message I received:

Error in trainNMF(x = sce, y = spe, groups = sce$type, mgs = mgs, weight_id = "weight",  : 
  ids %in% names(mgs) are not all TRUE
In addition: Warning message:
In max(table(mgs[[group_id]])) :
  no non-missing arguments to max; returning -Inf

Could you advise on how to resolve this issue?

I look forward to your response. Thank you!

**Comment by Lihyunseo** [comment]
I followed only your vignettes/SPOTlight_kidney.Rmd!

**Comment by 532at** [comment]
Hello, did you get any response? I encountered the same error, but I see most of issues are not answered? Perhaps, nobody is no longer working on that software?

**Comment by Lihyunseo** [comment]
Thank you for the email. Unfortunately, I haven't been able to solve the problem either. 😢
However, there are many other good tools available. How about comparing it with some of them?


-----Original Message-----

**Comment by 532at** [comment]
Yes, I did try CARD and RCTD which worked for me. Currently, trying to work with cell2location and STdeconvolve. :)
How about you?

**Comment by Lihyunseo** [comment]
Thank you for your reply! I'm preparing a presentation for my class comparing the results of STdeconvolve and SPOTlight.
For SPOTlight, the previous version works well.
You can find it here: https://github.com/MarcElosua/SPOTlight/tree/spotlight-0.1.7/R
I've heard that in R, RCTD is good, and in Python, cell2location is recommended.
I'm an undergraduate student at Soongsil University in Seoul, Korea. Who are you?

-----Original Message-----

**Comment by 532at** [comment]
Yes, but you can run cell2location and take cell type proportions to run downstream analysis in R - working on that currently. I am a graduate student (MD) doing my PhD. Currently, I am working in US, but I am from Poland. Happy to connect on LinkedIn (https://www.linkedin.com/in/adrianperdyan/)

**Comment by Lihyunseo** [comment]
Thank you for your reply I will go into the link you sent me!


-----Original Message-----

**Comment by gaohaohoa** [comment]
Here are some suggestions that may be worth trying
1. Ensure that the cell type is a string: `groups = as.character(sce$type)`
2. Please check if the following parameters are in the column names of `mgs`.
```
weight_id = "weight",
group_id = "type",
gene_id = "gene"
```
If I remember correctly, in the original `mgs`, I did not see "weight", but "mean.AUC", and group is "cluster", so maybe you can try my codes, however, which runs SLOWLY :
res <- SPOTlight(
  x = reference, 
  y = spatial_sce, 
  groups = as.character(reference$cell.types), 
  mgs = mgs_df, 
  scale = TRUE, 
  n_top = 10, 
  min_prop = 0.1, 
  hvg = high_variable_gene,
  weight_id = "mean.AUC",
  group_id = "cluster",
  gene_id = "gene",
  verbose = TRUE)

GOOD LUCK!!!!

**Comment by ouyangzhong** [comment]
I also encountered the same problem
